src/brogue/IO.c

```diff
@@ -2305,8 +2305,8 @@ void nextBrogueEvent(rogueEvent *returnEvent, boolean textInput, boolean colorsD
                 if (pauseDuration && pauseBrogue(pauseDuration)) {
                     // if the player did something during playback
                     nextBrogueEvent(&recordingInput, false, false, true);
-                    executePlaybackInput(&recordingInput);
-                    repeatAgain = !rogue.playbackPaused;
+                    interaction = executePlaybackInput(&recordingInput);
+                    repeatAgain = !rogue.playbackPaused && interaction;
```

Comment:
The logic of this loop is confusing me. We repeat the check for user input if the playback is unpaused and we received a "valid" input. Should this not be the opposite? The playing of the recording happens after this do-while loop so if it's unpaused surely we should drop out. Clearly I'm wrong but I don't know how :smile:

Reply:
The logic is really byzantine since recordings are played back through a combination of this function nextBrogueEvent(), mainBrogueJunction() and mainInputLoop().
I believe the logic is something like:
- When the recording is playing, pause after every move (pauseDuration) and give the use a chance to interact.
- If the interaction causes the recording to become paused after the interaction (e.g. skipping to a particular turn number) then drop out of the recording playback. The user can interact normally through mainInputLoop() [via pausePlayback() in mainBrogueJunction()] until the playback is restarted.
- Originally: if the user did an interaction that doesn't cause the playback to be paused, pause again in case they want to issue a second or subsequent command before the next move in the recording.
- My version: Only do this if they actually used a material command the first time. If they didn't just run onto the next move in the recording.